src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts

```diff
@@ -5,16 +5,27 @@
 
 import * as vscode from 'vscode'
 import * as fs from 'fs-extra'
-import { DependencyGraph } from '../dependencyGraph/dependencyGraph'
+import path = require('path')
 import { BM25Document, BM25Okapi } from './rankBm25'
-import { isRelevant } from './editorFilesUtil'
 import { ToolkitError } from '../../../shared/errors'
 import { UserGroup, crossFileContextConfig, supplemetalContextFetchingTimeoutMsg } from '../../models/constants'
 import { CancellationError } from '../../../shared/utilities/timeoutUtils'
-import { CodeWhispererSupplementalContextItem } from './supplementalContextUtil'
+import { CodeWhispererSupplementalContextItem, getOpenFilesInWindow } from './supplementalContextUtil'
 import { CodeWhispererUserGroupSettings } from '../userGroupUtil'
+import { isTestFile } from './codeParsingUtil'
+import * as CodeWhispererConstants from '../../models/constants'
+
+// TODO: ugly, can we make it prettier?
+// TODO: Move to another config file or constants file
+// Supported language to its dialects
+const supportedLanguageToDialects: Record<string, Set<string>> = {
```

Comment:
const supportedLanguageToDialects = {
         CodeWhispererConstants.java: new Set<string>(CodeWhispererConstants.java)

}

You don't have to use Record

Reply:
Better implement a singleton class for cross file context.

class CrossFileLanguageUtil {
       const supportedLanguageToDialects: Record<string, Set<string>> = {}

      isSupportedLanguage(): boolean {
      }
      
      
}

Use this class to control 
1. which language is supported for all
2. which dialect is supported
3. which language  is supported only for user group

Reply:
In general, aim for a high cohesion code

Reply:
yea agree, I prefer restructuring the whole into more OOP style, but it might take time

Reply:
will leave it as is now @leigaol

Reply:
I aim to abstract the whole crossfile & utg into a single `supplementalContextProvider` or do it seperately `crossfileContextProvider` `utgContextProvider`

Reply:
ACK. By when do you need this merged? I can approve once blocking comments are resolved.